mbofui/app/js/controllers.js

```diff
@@ -271,6 +267,25 @@ ocellus.controller('mapController', ['$compile', '$scope', '$rootScope','$filter
     }
   });
 
+  $scope.removeEvent = function(event){
+    // get position of this event in the 3 collections
+    var thisEvent = _.findWhere($rootScope.events, {url: event.url});
+    var thisEventPos = $rootScope.events.indexOf(_.findWhere($rootScope.events, {url: event.url}));
+    var thisEventPosMF = $scope.markersFiltered.indexOf(_.findWhere($rootScope.events, {url: event.url}));
+    var thisEventPosM = $scope.markersAll.indexOf(_.findWhere($rootScope.events, {url: event.url}));
+    // remove the event from the three collections
+    $rootScope.events.splice(thisEventPos, 1);
+    $scope.markersFiltered.splice(thisEventPosMF, 1);
+    $scope.markersAll.splice(thisEventPosM, 1);
+  };
+
+  $scope.eventEditCancel = function(){
+    $scope.editEvent.status = 'cancelled';
+    $scope.removeEvent($scope.editEvent);
+    $scope.eventEditPost();
+
+  };
+
   // handler for event edit PUT
   $scope.eventEditPost = function (){
     // use the editEvent model for the data for this PUT
```

Comment:
The Webshim polyfill  for datetime is expecting a string with a particular format. Angular demands a  js Date object for datetime input - so we cannot do it the Angular way - that is why we populate the datetime inputs with jQuery and retrieve the edited values with jQuery. If for some reason the user removes all the values we fall back to the angular scope values of the event (the previous values)